### PR TITLE
Fix rsyslog plugin slim image logic

### PR DESCRIPTION
**Pull request by zbud-msft**
##### Work item tracking
- Microsoft ADO **(number only)**: 27773623

#### How I did it

Added checks after redirect to frr/bgpd.log

#### How to verify it

Manual test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: files/image_config/rsyslog/rsyslog.d/00-sonic.conf.j2 [hunk 1]
```diff
@@ -10,26 +10,35 @@ template(name="prog_msg" type="list") {
 
 $ModLoad omprog
 
+{% endif %}
+{% endif %}
+
 if re_match($programname, "bgp[0-9]*#(frr|zebra|staticd|watchfrr)") then {
     /var/log/frr/zebra.log
+    {% if include_system_eventd == "y" %}
+    {% if build_reduce_image_size != "y" or sonic_asic_platform != "broadcom" %}
```
Comment (saiarcot895) [inline files/image_config/rsyslog/rsyslog.d/00-sonic.conf.j2:19]: Why have a ASIC check for Broadcom here?

Comment (zbud-msft) [reply files/image_config/rsyslog/rsyslog.d/00-sonic.conf.j2:19]: eventd should not be enabled on 7050qx and 7060cx3, so we check flags build_reduce_image_size for slim image enabled and also for broadcom asic.

Comment (qiluo-msft) [reply files/image_config/rsyslog/rsyslog.d/00-sonic.conf.j2:19]: Why single flag is not enough? for example `build_reduce_image_size`?

Comment (zbud-msft) [reply files/image_config/rsyslog/rsyslog.d/00-sonic.conf.j2:19]: Initial requirement was for disabling eventd in 7050qx/7060 which is why I added a tighter check.

**Review by qiluo-msft — APPROVED** [review]

**Comment by StormLiangMS** [comment]
@yxieca  for 202311 cherry pick, this one would be nice to have for next image build.

**Comment by TafkaMax** [comment]
I am building for broadcom and I got this error:

It seems to be related? eventd is the codeword here.

```
Assertion failed: _term_acks > 0 (src/own.cpp:180)
make[2]: *** [Makefile:60: eventd-tests] Aborted (core dumped)
make[2]: *** Waiting for unfinished jobs....
Finished building: rsyslog_plugin_tests/rsyslog_plugin_ut.cpp
 
make[2]: Leaving directory '/sonic/src/sonic-eventd'
dh_auto_build: error: make -j8 "INSTALL=install --strip-program=true" returned exit code 2
make[1]: *** [debian/rules:6: build] Error 25
make[1]: Leaving directory '/sonic/src/sonic-eventd'
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
[  FAIL LOG END  ] [ target/debs/bullseye/sonic-eventd_1.0.0-0_amd64.deb ]
make: *** [slave.mk:775: target/debs/bullseye/sonic-eventd_1.0.0-0_amd64.deb] Error 1
make: *** Waiting for unfinished jobs....
[ finished ] [ target/python-wheels/bullseye/sonic_py_common-1.0-py3-none-any.whl ] 
[ finished ] [ target/debs/bullseye/libsairedis_1.0.0_amd64.deb ] 
make[1]: *** [Makefile.work:600: bullseye] Error 2
make[1]: Leaving directory '/home/gitlab-runner/builds/EzyZQ4s5/0/sysadm/sonic-builder/sonic-buildimage'
make: *** [Makefile:46: target/sonic-broadcom.bin] Error 2
```

**Comment by zbud-msft** [comment]
@TafkaMax This change should not affect build of sonic-eventd deb package.

**Comment by TafkaMax** [comment]
> @TafkaMax This change should not affect build of sonic-eventd deb package.

Yes, it builds now, must have been something else. Sorry for bothering.

**Pull request merged**